api/api.yaml

```diff
@@ -153,19 +158,36 @@ definitions:
       es_field_type:
         type: string
         description: The Elasticsearch field type.
-      values:
+      value_names:
         type: array
         items:
-          $ref: "#/definitions/FacetValue"
-  FacetValue:
-    description: A facet value, for example, "Male"/34.
-    properties:
-      name:
-        type: string
-        description: Facet value name, for example, "Male".
-      count:
+          type: string
+        description: Array of names of possible facet values.
+      # Either values should be set, or both of time_names and
+      # time_series_value_counts should be set, but not all three.
+      value_counts:
+        type: array
+        items:
+          type: integer
+        description: Array of counts for each facet value.
+      time_names:
+        type: array
+        items:
+          type: integer
+        description: Array of times.
+      time_series_value_counts:
+        type: array
+        items:
+          type: array
+          items:
+            type: integer
+        description: >
+          2-dimensional array of facet value counts, indexed by time
+          then value; indexes correspond to time_names and then
+          value_names.
+      is_time_series:
```

Comment:
What do you think about deleting this and checking if time_names or time_series_value_counts is set instead? I think that's cleaner.

For Elasticsearch, IIRC there wasn't an easy way to tell if a field was time series; that's why we added _is_time_series field

Reply:
The only case where is_time_series is needed here is when the facet has no data, so that value_names, value_counts, time_names, and time_series_value_counts are all empty. The current implementation of empty time series facets doesn't show the zero count in the upper right corner, whereas non-time series facets do. (The heatmap also had empty time series facets be twice as wide). Do you think it's cleaner just to get rid of these differences, or else explain this in the is_time_series description?

Reply:
I prefer to not have `is_time_series` here. I don't think we even need to mention the "no data" case in these comments... for time series facets with no data, I don't have a strong opinion on whether we should show zero in upper right or not.

Reply:
Ok, will remove.